x-pack/plugins/enterprise_search/public/applications/shared/cypress/tsconfig.json

```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../../../../../../../tsconfig.base.json",
+  "exclude": [],
+  "include": ["./**/*"],
+  "compilerOptions": {
+    "types": ["cypress", "node"]
+  }
```

Comment:
References as to why the separate tsconfig JSONs are necessary:

- https://docs.cypress.io/guides/tooling/typescript-support#Clashing-types-with-Jest
- Example of Security team's `cypress/tsconfig.json`: https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/cypress/tsconfig.json (we may need more complex settings like theirs at some point, but for now this is all I needed to get started)

Reply:
@mistic @spalger, I am trying to wrap my head around this will affect publishing the plugin (server, common and public) TS interfaces as this now introduces a nested tsconfig.

Reply:
I'm not super clear either personally, but for what it's worth I added an `exclude` option in our main tsconfig:

https://github.com/elastic/kibana/blob/1b3ea89fc531f8e8d75949d3c2ebb311b6f9721f/x-pack/plugins/enterprise_search/tsconfig.json#L9-L14

So I at least for us, I _think_ that makes it not nested? Not sure about the Security solution's setup though